### PR TITLE
Add token types

### PR DESCRIPTION
**Pull request by rwblickhan**
:tickets: **Ticket(s)**: Affects #13

---

## :construction_worker: Changes

Added a struct to encapsulate an individual token, as well as an enum to represent the various token types we will need.

## :flashlight: Testing Instructions

No testing necessary.

### CODE REVIEW

File: src/include/Token.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+enum struct TokenType {
+    LeftParen, RightParen, LeftBrace, RightBrace,
```
Comment (bobheadxi) [inline src/include/Token.h]: These look grouped - maybe use comments to delimit different "groups" of tokens (`delimiters`, `operators`, etc.) for the sake of readability?

**Review by bobheadxi — CHANGES_REQUESTED** [review]
Looks more or less good, just a thing on documenting the token types

File: src/include/Token.h (hunk 1 above)
Comment (rwblickhan) [reply src/include/Token.h]: Haha, I actually had that originally but removed it because it looked a bit odd to me. I can add it back though.

Comment (bobheadxi) [reply src/include/Token.h]: I think it shouldn't look too weird 👍 

```cpp
enum struct TokenType {
   // ...something about a group
   LeftParen, RightParen, LeftBrace, RightBrace,
   
   // ...next group
   bla, bla, bla

}
```

Comment (rwblickhan) [reply src/include/Token.h]: I think it's the extra space between the groups that makes it work---it looks really dense otherwise. Anyway, updated.

**Review by bobheadxi — APPROVED** [review]
Looks great! 🔥

**Comment by rwblickhan** [comment]
In the latest revision I also removed the Dot TokenType. I'm not sure how that snuck in there but (I think?) we're not using `.` for anything at the moment.

**Comment by bobheadxi** [comment]
Right, sounds good to me 🆗

**Pull request merged**